src/read/dwarf.rs

```diff
@@ -297,6 +297,12 @@ impl<R: Reader> Dwarf<R> {
         unit: &Unit<R>,
         offset: RangeListsOffset<R::Offset>,
     ) -> Result<RngListIter<R>> {
+        let offset = if self.file_type == DwarfFileType::Dwo && unit.header.version() < 5 {
+            RangeListsOffset(offset.0.wrapping_add(unit.rnglists_base.0))
```

Comment:
This means that `RangeListsOffset` has different meanings at different places in the code. While your changes look correct, it is hard to be sure, and ditto for users of this library. e.g. it looks like the `AttributeValue::RangeListsRef` support in `write/unit.rs` needs fixing too. I prefer if we introduce something like `RawRangeListsOffset` that is used in `AttributeValue`, and then convert that into a `RangeListsOffset`. Or do the conversion when parsing attributes but we may not have the required info there.

Reply:
Yeah, the downside of a new type is that then everybody has to be aware of and handle this stuff that's only needed for a non-standard and obsolete extension.

I've never looked at the writing side of gimli but if it's at all possible I would just not support writing these non-standard GNU extensions.

Reply:
That's a valid concern, but if it's important enough for you to handle it, then maybe others want to as well? It is uglier to use, but see what you think of https://github.com/philipc/gimli/commit/473e138d736bb6ed0b2515a5160e6f3385b99ac3

On the writing side, the only support required is for reading it when converting. It won't write it out in the same form.